src/tests.rs

```diff
@@ -0,0 +1,147 @@
+                }
+
+                let event_timestamp: &'static str =
+                    Box::leak(time_since_epoch.to_string().into_boxed_str());
+
+                test_event.id = Some(test_uuid.clone());
+                let now = SnowplowTimestamp::now();
+                let events = [test_event].into_iter().map(|event| SnowplowEvent {
+                    event_type: EventType::SelfDescribingEvent,
+                    payload: JsonString(PayloadWrapper::new(event.payload)),
+                    platform: Platform::Desktop,
+                    app_id: "test id",
+                    tracker_id: "test tracker ID",
+                    namespace: "test namespace",
+                    event_id: event.id,
+                    created_timestamp: event.timestamp.unwrap_or(now),
+                    sent_timestamp: event.timestamp.unwrap_or(now),
+                });
+
+                let events = EventContainer::new(events);
+                assert_ser_tokens(
+                    &events.readable(),
+                    &[
+                        Token::Struct {
+                            name: "Envelope",
+                            len: 2,
+                        },
+                        Token::Str("schema"),
+                        Token::Str("iglu:com.snowplowanalytics.snowplow/payload_data/jsonschema/1-0-4"),
+                        Token::Str("data"),
+                        Token::Seq { len: Some(1), },
+                        Token::Struct { name: "SnowplowEvent", len: 9, },
+                        Token::Str("e"),
+                        Token::UnitVariant { name: "EventType", variant: "ue", },
+                        Token::Str("ue_pr"),
+                        Token::Str("{\"schema\":\"iglu:com.snowplowanalytics.snowplow/unstruct_event/jsonschema/1-0-0\",\"data\":{\"schema\":\"iglu:com.snowplowanalytics.snowplow/screen_view/jsonschema/1-0-0\",\"data\":{\"name\":\"test\",\"id\":\"test id\"}}}"),
+                        Token::Str("p"),
+                        Token::UnitVariant { name: "Platform", variant: "pc", },
+                        Token::Str("aid"),
+                        Token::Str("test id"),
+                        Token::Str("tv"),
+                        Token::Str("test tracker ID"),
+                        Token::Str("tna"),
+                        Token::Str("test namespace"),
+                        Token::Str("eid"),
+                        Token::Some,
+                        Token::Str("a1a2a3a4-b1b2-c1c2-d1d2-d3d4d5d6d7d8"),
+                        Token::Str("dtm"),
+                        Token::Str(event_timestamp),
+                        Token::Str("stm"),
+                        Token::Str(event_timestamp),
+                        Token::StructEnd,
+                        Token::SeqEnd,
+                        Token::StructEnd,
+                    ]
+                );
+            }
+            Err(_) => println!("Something went wrong when parsing the UUID"),
```

Comment:
Instead of using a match, do `let test_uuid = Uuid::parse_str("a1a2a3a4b1b2c1c2d1d2d3d4d5d6d7d8").expect("failed to create UUID")`. This will panic if there's an error, and panicking is the main way to fail tests